YAML changelogs with many references cannot be parsed with snakeyaml 1.26

<!--- This environment context section helps us quickly review your PR. 
      Please take a minute to fill-out this information. -->

## Environment
**Liquibase Version**:  3.8.9

**Liquibase Integration & Version**: Spring Boot 2.3.3

**Liquibase Extension(s) & Version**: N/A

**Database Vendor & Version**: hsqldb (but this is not relevant)

**Operating System Type & Version**: MacOS 10.13.6 (also not relevant)

## Description
The current version of snakeyaml has several [configurable security features](https://github.com/asomov/snakeyaml/commit/da11ddbd91c1f8392ea932b37fa48110fa54ed8c) to guard against a billion-laughs attack. Unfortunately, [one of them](https://github.com/asomov/snakeyaml/blob/da11ddbd91c1f8392ea932b37fa48110fa54ed8c/src/main/java/org/yaml/snakeyaml/composer/Composer.java#L146) also guards against the existing (previously working) [changelog](https://github.com/usds/case-issue-api/blob/f4f60b03f6df6bbd56d9f11778fc514a684b3dc0/src/main/resources/db/changelog/db.changelog-master.yaml) for the project I'm trying to upgrade (from liquibase 3.6.3, snakeyaml 1.23 and Spring Boot 2.1.15), which uses collection aliases for boilerplate in column definitions. This means that when liquibase runs during Spring context startup, it fails with the error `Number of aliases for non-scalar nodes exceeds the specified max=50`. 

This maximum value is configurable in snakeyaml, but the configuration is not exposed by liquibase, which means that there does not appear to be a way to work around this issue. The obvious place to inject the configuration is this call to [SafeConstructor()](https://github.com/liquibase/liquibase/blob/master/liquibase-core/src/main/java/liquibase/parser/core/yaml/YamlChangeLogParser.java#L23), which could instead be a call to `SafeConstructor(loadingProperties)`. It's not obvious (to me, anyway) how to initialize that LoadingProperties object, however.

The changelog that produces this error is, conveniently, part of an open-source project, so you can see the alias definitions starting [on line 7](https://github.com/usds/case-issue-api/blob/master/src/main/resources/db/changelog/db.changelog-master.yaml#L7), and the uses of them in a lot of table definitions below.

## Steps To Reproduce
* `git clone https://github.com/usds/case-issue-api && cd case-issue-api`
* `git checkout ben-warfield/liquibase-snakeyaml-crash`
* `SPRING_PROFILES_ACTIVE=db-migrate ./gradlew bootRun`

A more compact reproduction case using the liquibase CLI is presumably possible, but this is pretty easy as long as you don't mind downloading a bunch of jars.

* https://github.com/usds/case-issue-api/blob/master/src/main/resources/db/changelog/db.changelog-master.yaml#L7
* Exact commands used - CLI, maven, gradle, spring boot, servlet, etc.

## Actual Behavior
```
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'liquibase' defined in class path resource [org/springframework/boot/autoconfigure/liquibase/LiquibaseAutoConfiguration$LiquibaseConfiguration.class]: Invocation of init method failed; nested exception is liquibase.exception.ChangeLogParseException: Syntax error in file classpath:/db/changelog/db.changelog-master.yaml: Number of aliases for non-scalar nodes exceeds the specified max=50
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1794)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:594)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:516)
        at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:324)
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:226)
        at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:322)
        at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202)
        at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:311)
        at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202)
        at org.springframework.context.support.AbstractApplicationContext.getBean(AbstractApplicationContext.java:1109)
        at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:869)
        at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:551)
        at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:758)
        at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:750)
        at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:397)
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:315)
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1237)
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1226)
        at gov.usds.case_issues.CaseIssueApi.main(CaseIssueApi.java:14)
Caused by: liquibase.exception.ChangeLogParseException: Syntax error in file classpath:/db/changelog/db.changelog-master.yaml: Number of aliases for non-scalar nodes exceeds the specified max=50
        at liquibase.parser.core.yaml.YamlChangeLogParser.parseYamlStream(YamlChangeLogParser.java:92)
        at liquibase.parser.core.yaml.YamlChangeLogParser.parse(YamlChangeLogParser.java:31)
        at liquibase.Liquibase.getDatabaseChangeLog(Liquibase.java:217)
        at liquibase.Liquibase.update(Liquibase.java:190)
        at liquibase.Liquibase.update(Liquibase.java:179)
        at liquibase.integration.spring.SpringLiquibase.performUpdate(SpringLiquibase.java:366)
        at liquibase.integration.spring.SpringLiquibase.afterPropertiesSet(SpringLiquibase.java:314)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1853)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1790)
        ... 18 common frames omitted
Caused by: org.yaml.snakeyaml.error.YAMLException: Number of aliases for non-scalar nodes exceeds the specified max=50
        at org.yaml.snakeyaml.composer.Composer.composeNode(Composer.java:147)
        at org.yaml.snakeyaml.composer.Composer.composeValueNode(Composer.java:257)
        at org.yaml.snakeyaml.composer.Composer.composeMappingChildren(Composer.java:248)
        at org.yaml.snakeyaml.composer.Composer.composeMappingNode(Composer.java:236)
        at org.yaml.snakeyaml.composer.Composer.composeNode(Composer.java:162)
        at org.yaml.snakeyaml.composer.Composer.composeSequenceNode(Composer.java:209)
        at org.yaml.snakeyaml.composer.Composer.composeNode(Composer.java:160)
        at org.yaml.snakeyaml.composer.Composer.composeValueNode(Composer.java:257)
        at org.yaml.snakeyaml.composer.Composer.composeMappingChildren(Composer.java:248)
        at org.yaml.snakeyaml.composer.Composer.composeMappingNode(Composer.java:236)
        at org.yaml.snakeyaml.composer.Composer.composeNode(Composer.java:162)
        at org.yaml.snakeyaml.composer.Composer.composeValueNode(Composer.java:257)
        at org.yaml.snakeyaml.composer.Composer.composeMappingChildren(Composer.java:248)
        at org.yaml.snakeyaml.composer.Composer.composeMappingNode(Composer.java:236)
        at org.yaml.snakeyaml.composer.Composer.composeNode(Composer.java:162)
        at org.yaml.snakeyaml.composer.Composer.composeSequenceNode(Composer.java:209)
        at org.yaml.snakeyaml.composer.Composer.composeNode(Composer.java:160)
        at org.yaml.snakeyaml.composer.Composer.composeValueNode(Composer.java:257)
        at org.yaml.snakeyaml.composer.Composer.composeMappingChildren(Composer.java:248)
        at org.yaml.snakeyaml.composer.Composer.composeMappingNode(Composer.java:236)
        at org.yaml.snakeyaml.composer.Composer.composeNode(Composer.java:162)
        at org.yaml.snakeyaml.composer.Composer.composeValueNode(Composer.java:257)
        at org.yaml.snakeyaml.composer.Composer.composeMappingChildren(Composer.java:248)
        at org.yaml.snakeyaml.composer.Composer.composeMappingNode(Composer.java:236)
        at org.yaml.snakeyaml.composer.Composer.composeNode(Composer.java:162)
        at org.yaml.snakeyaml.composer.Composer.composeSequenceNode(Composer.java:209)
        at org.yaml.snakeyaml.composer.Composer.composeNode(Composer.java:160)
        at org.yaml.snakeyaml.composer.Composer.composeValueNode(Composer.java:257)
        at org.yaml.snakeyaml.composer.Composer.composeMappingChildren(Composer.java:248)
        at org.yaml.snakeyaml.composer.Composer.composeMappingNode(Composer.java:236)
        at org.yaml.snakeyaml.composer.Composer.composeNode(Composer.java:162)
        at org.yaml.snakeyaml.composer.Composer.getNode(Composer.java:95)
        at org.yaml.snakeyaml.composer.Composer.getSingleNode(Composer.java:119)
        at org.yaml.snakeyaml.constructor.BaseConstructor.getSingleData(BaseConstructor.java:150)
        at org.yaml.snakeyaml.Yaml.loadFromReader(Yaml.java:472)
        at org.yaml.snakeyaml.Yaml.load(Yaml.java:411)
        at liquibase.parser.core.yaml.YamlChangeLogParser.parseYamlStream(YamlChangeLogParser.java:90)
        ... 26 common frames omitted
```

## Expected/Desired Behavior
This changelog should be able to be parsed—there's nothing particularly hazardous about this use of YAML features, it's just hitting a circuit breaker. I would argue that setting the limit to an arbitrarily high value is safe in the context of liquibase changelogs, since if a malicious actor has access to your database changelog, crashing the parser is really the least of your worries, but at the very least, it should be possible to set a system property to configure it to a larger number on a case-by-case value (and also, I would imagine, to do the other configurations allowed by [LoaderOptions](https://github.com/asomov/snakeyaml/blob/master/src/main/java/org/yaml/snakeyaml/LoaderOptions.java).

--- a/liquibase-standard/src/main/java/liquibase/parser/core/yaml/YamlParser.java
+++ b/liquibase-standard/src/main/java/liquibase/parser/core/yaml/YamlParser.java
@@ -15,6 +15,7 @@ public abstract class YamlParser implements LiquibaseParser {
     public static LoaderOptions createLoaderOptions() {
         LoaderOptions options = new LoaderOptions();
         SnakeYamlUtil.setCodePointLimitSafely(options, Integer.MAX_VALUE);
+        SnakeYamlUtil.setMaxAliasesForCollections(options, Integer.MAX_VALUE);
         SnakeYamlUtil.setProcessCommentsSafely(options, false);
         // TODO: remove the below line when we have a general fix for the not allowed duplicated databaseChangelog and sql tags
         //        options.setAllowDuplicateKeys(false);

--- a/liquibase-standard/src/main/java/liquibase/util/SnakeYamlUtil.java
+++ b/liquibase-standard/src/main/java/liquibase/util/SnakeYamlUtil.java
@@ -39,6 +39,13 @@ public class SnakeYamlUtil {
     }
 
     /**
+     * Safely set the limit for maxAliasesForCollections when using aliases for repetitive columns while configuring a new SnakeYaml instance.
+     */
+    public static void setMaxAliasesForCollections(LoaderOptions loaderOptions, int maxAliasesForCollections) {
+        safelyCallNewSnakeYamlMethod(() -> loaderOptions.setMaxAliasesForCollections(maxAliasesForCollections));
+    }
+
+    /**
      * Helper method to make sure that we display the error message only once.
      */
     private static void safelyCallNewSnakeYamlMethod(Runnable code) {

